toucan_connectors/snowflake_oauth2/snowflake_oauth2_connector.py

```diff
@@ -90,69 +116,96 @@ def build_authorization_url(self, **kwargs):
-            'token': self.get_access_token(),
-            'role': self.role if self.role else '',
-        }
-        token_end = timer()
-        logger.info(
-            f'[benchmark] - get_access_token {token_end - token_start} seconds',
-            extra={
-                'benchmark': {
-                    'operation': 'get_access_token',
-                    'execution_time': token_end - token_start,
-                }
-            },
-        )
-        return pysnowflake.connector.connect(
-            **connection_params, database=database, warehouse=warehouse
+    def _get_connection(self, database: str = None, warehouse: str = None) -> SnowflakeConnection:
+        def connect_function() -> SnowflakeConnection:
+            logger.info('Connect at Snowflake')
+            token_start = timer()
+            tokens = self.get_access_token()
+            token_end = timer()
+
+            connection_params = {
+                'account': self.account,
+                'authenticator': AuthenticationMethod.OAUTH,
+                'application': 'ToucanToco',
+                'token': tokens,
+                'role': self.role if self.role else '',
+            }
+            logger.info(
+                f'[benchmark] - get_access_token {token_end - token_start} seconds',
+                extra={
+                    'benchmark': {
+                        'operation': 'get_access_token',
+                        'execution_time': token_end - token_start,
+                    }
+                },
+            )
+
+            logger.info(
+                f'Connect at Snowflake with {connection_params}, database {database} and warehouse {warehouse}'
+            )
+            connect_start = timer()
+            c = snowflake.connector.connect(
+                **connection_params, database=database, warehouse=warehouse
+            )
+            connect_end = timer()
+            logger.info(
+                f'[benchmark] - connect {connect_end - connect_start} seconds',
+                extra={
+                    'benchmark': {
+                        'operation': 'connect',
+                        'execution_time': connect_end - connect_start,
+                    }
+                },
+            )
+            return c
+
+        def alive_function() -> Any:
+            logger.info('Check Snowflake connection')
+            if hasattr(connection, 'is_closed') and callable(connection.is_closed):
+                return connection.is_closed()
+
+        def close_function() -> None:
+            logger.info('Close Snowflake connection')
+            if hasattr(connection, 'close') and callable(connection.close):
+                return connection.close()
+
+        connection: SnowflakeConnection = connection_manager.get(
+            self.identifier,
+            connect_method=lambda: connect_function(),
+            alive_method=lambda: alive_function(),
+            close_method=lambda: close_function(),
         )
```

Comment:
On passe une lambda ou directement la référence de la méthode ? 
```suggestion
            connect_method=connect_function,
            alive_method=alive_function,
            close_method=close_function,
```